sample_project/Assets/SampleViewer/Scripts/SampleSwitcher.cs

```diff
@@ -12,36 +12,46 @@
 using UnityEngine.SceneManagement;
 using UnityEngine.UI;
 using Esri.ArcGISMapsSDK.Utils;
+using TMPro;
 
 
 public class SampleSwitcher : MonoBehaviour
 {
-	public GameObject MenuVideo;
-	public Button ExitButton;
-	public Camera cam;
-	public Button[] sceneButtons;
-	public Button[] pipelineButtons;
-	private string apiKey;
-	private string projectAPIKey;
+    [SerializeField] private string apiKey;
+    [SerializeField] private TextMeshProUGUI apiKeyInputField;
+    [SerializeField] private GameObject notificationMenu;
+    [SerializeField] private GameObject warning;
+
+    private Animator animator;
+    private string currentSceneName;
+    private string nextSceneName;
+    private string projectAPIKey;
 	private string pipelineModeText;
-	private string nextSceneName;
 	private string PipelineType;
-	private string currentSceneName;
 	private int sceneLoadedCount;
-	private Animator animator;
 
-	[SerializeField] private GameObject warning;
-	[SerializeField] private GameObject notificationMenu;
+    public GameObject MenuVideo;
+    public Button ExitButton;
+    public Camera cam;
+    public Button[] sceneButtons;
+    public Button[] pipelineButtons;
 
-	private void Start()
+    private void Start()
 	{
 		animator = notificationMenu.GetComponent<Animator>();
 		cam.enabled = true;
 
 		Invoke("SlideNotification", 2.0f);
 
-		projectAPIKey = ArcGISProjectSettingsAsset.Instance.APIKey;
-		CheckAPIKey(projectAPIKey);
+		if (apiKey != null && apiKey.Length == 100)
```

Comment:
Do we know if that length is always 100? Also why not check for these things in the CheckAPIKey method, then return a bool?

Reply:
I was just following how it's currently being checked which is checking if it's of length 100

Reply:
I think this should be apart of the Sample viewer refactor issue. Trying to fix this is resulting in either the project settings API key no longer working to unlock samples or the input field API key no longer working to unlock samples.